allennlp/data/dataset_readers/vqav2.py

```diff
@@ -533,3 +530,35 @@ def text_to_instance(
     @overrides
     def apply_token_indexers(self, instance: Instance) -> None:
         instance["question"].token_indexers = self._token_indexers  # type: ignore
+
+    def _get_answers_by_question_id(self, split):
+        answers_by_question_id = {}
+        if split.annotations is not None:
+            # Pre-processing the annotations is time-consuming, so we don't want to
+            # have to re-do it each time we call read(). So we cache this result.
+            annotations_path = cached_path(split.annotations, extract_archive=True)
+            with LocalCacheResource(split.annotations + "-cache", annotations_path) as cache:
```

Comment:
What's this? I missed this the first time around. This calculation used to take seconds. Now building the cache takes minutes. Why did you cache this?

Reply:
It takes 2 minutes on my machine now (without writing the cache). That's certainly odd. Not sure why this started.

Reply:
With a4baae6ed4431c019e27c2a1719cd308669fabda, it takes 15 seconds on my machine to calculate them all. I think that's preferable to caching to disk.